src/features/slackapi/slackapi-utils.ts

```diff
@@ -107,30 +146,42 @@ Have a great week!
    * @param previousWorkDays dates and the number of today
    * @param nonProjectTimes all non project times
    */
-  export const postDailyMessage = (
+  export const postDailyMessage = async (
     dailyCombinedData: DailyCombinedData[],
     timeRegistrations: TimeRegistrations[],
     previousWorkDays: PreviousWorkdayDates,
-    nonProjectTimes: NonProjectTime[]) => {
-    const { numberOfToday, yesterday, today } = previousWorkDays;
-
-    dailyCombinedData.forEach(user => {
-      const { slackId, personId, expected } = user;
-
-      const isAway = TimeUtilities.checkIfUserIsAwayOrIsItFirstDayBack(timeRegistrations, personId, expected, today, nonProjectTimes);
-      const firstDayBack= TimeUtilities.checkIfUserIsAwayOrIsItFirstDayBack(timeRegistrations, personId, expected, yesterday, nonProjectTimes);
-
-      if (!isAway && !firstDayBack) {
-        try {
-          client.chat.postMessage({
-            channel: slackId,
-            text: constructDailyMessage(user, numberOfToday)
-          });
-        } catch (error) {
-          console.error(`Error while posting slack messages to user ${user.name}`);
+    nonProjectTimes: NonProjectTime[]): Promise<DailyMessageData[]> => {
+    try{
+      const { numberOfToday, yesterday, today } = previousWorkDays;
+
+      let messagesRecord: DailyMessageData[] = [];
+
+      const promises = dailyCombinedData.map(user => {
+        const { slackId, personId, expected } = user;
+
+        const isAway = TimeUtilities.checkIfUserIsAwayOrIsItFirstDayBack(timeRegistrations, personId, expected, today, nonProjectTimes);
+        const firstDayBack= TimeUtilities.checkIfUserIsAwayOrIsItFirstDayBack(timeRegistrations, personId, expected, yesterday, nonProjectTimes);
+
+        const message = constructDailyMessage(user, numberOfToday);
+
+        if (!isAway && !firstDayBack && expected !== 0) {
+          console.log(message.message, slackId);
+          messagesRecord.push(message);
+          return sendMessage(slackId, message.message);
         }
+      });
+
+      const responses: ChatPostMessageResponse[] = await Promise.all(promises);
+      const errors = responses.filter(response => response?.error);
```

Comment:
This cannot be undefined.
```suggestion
      const errors = responses.filter(response => response.error);
```